python/build_template/configure_build.py

```diff
@@ -0,0 +1,349 @@
+    hdr_name = "MAIN"
+    if new_gen == False:
+        #pre rb2 does not use miloVersion for superfreq image generation on ps2
+        miloVersion = " "
+        #remove these two if rock band 1
+        ark_encrypt = " "
+        dtb_encrypt = dtb_encrypt.upper()
+
+ninja.variable("dtb_encrypt", dtb_encrypt)
+ninja.variable("ark_encrypt", ark_encrypt)
+ninja.variable("miloVersion", miloVersion)
+
+#new gen games (rb2 onward) add platform suffix to ark name
+if new_gen == True:
+    hdr_name = hdr_name + "_" + args.platform
+
+print(f"Configuring {game_name}...")
+print(f"Platform: {args.platform}")
+
+# configure tools
+ark_dir = Path("obj", args.platform, "ark")
+match sys.platform:
+    case "win32":
+        ninja.variable("silence", ">nul")
+        ninja.rule("copy", "cmd /c copy $in $out $silence", description="COPY $in")
+        ninja.rule("bswap", "dependencies\\windows\\swap_art_bytes.exe $in $out", description="BSWAP $in")
+        ninja.rule("version", "python dependencies\\python\\gen_version.py $out", description="Writing version info")
+        ninja.rule("png_list", "python dependencies\\python\\png_list.py $dir $out", description="PNGLIST $dir")
+        ninja.variable("superfreq", "dependencies\\windows\\superfreq.exe")
+        ninja.variable("arkhelper", "dependencies\\windows\\arkhelper.exe")
+        ninja.variable("dtab", "dependencies\\windows\\dtab.exe")
+        ninja.variable("dtacheck", "dependencies\\windows\\dtacheck.exe")
+    case "darwin":
+        ninja.variable("silence", "> /dev/null")
+        ninja.rule("copy", "cp $in $out", description="COPY $in")
+        ninja.rule("bswap", "python3 dependencies/python/swap_rb_art_bytes.py $in $out", description="BSWAP $in")
+        ninja.rule("version", "python3 dependencies/python/gen_version.py $out", description="Writing version info")
+        ninja.rule("png_list", "python3 dependencies/python/png_list.py $dir $out", description="PNGLIST $dir")
+        ninja.variable("superfreq", "dependencies/macos/superfreq")
+        ninja.variable("arkhelper", "dependencies/macos/arkhelper")
+        ninja.variable("dtab", "dependencies/macos/dtab")
+        # dtacheck needs to be compiled for mac
+        ninja.variable("dtacheck", "true")
+    case "linux":
+        ninja.variable("silence", "> /dev/null")
+        ninja.rule("copy", "cp --reflink=auto $in $out",description="COPY $in")
+        ninja.rule("bswap", "dependencies/linux/swap_art_bytes $in $out", "BSWAP $in")
+        ninja.rule("version", "python dependencies/python/gen_version.py $out", description="Writing version info")
+        ninja.rule("png_list", "python dependencies/python/png_list.py $dir $out", description="PNGLIST $dir")
+        ninja.variable("superfreq", "dependencies/linux/superfreq")
+        ninja.variable("arkhelper", "dependencies/linux/arkhelper")
+        ninja.variable("dtab", "dependencies/linux/dtab")
+        ninja.variable("dtacheck", "dependencies/linux/dtacheck")
```

Comment:
this is me being stupid but there's probably a much cleaner way to rewrite this. i'll look into this when i redo the rb3dx script if you want to pass that job off to me